File: 프로그래머스/lv2/42839. 소수 찾기/소수 찾기.py
import math 
from itertools import permutations

def is_prime(n):
    if n==0 or n==1:
        return False
    else:
        for i in range(2, int(math.sqrt(n))+1):
            if n % i ==0:
                return False 
        return True
    
def solution(numbers):
    ans = []
    temp = []
    for i in range(1,len(numbers)+1):
        arr = permutations(list(numbers),i)
        for j in arr:
            ans.append(int(''.join(j)))
    
    ans = list(set(ans))
    for i in range(len(ans)):
        print(ans[i])
        if is_prime(int(ans[i])):
            temp.append(int(ans[i]))
    
    return len(temp)

File: 프로그래머스/lv1/12921. 소수 찾기/소수 찾기.java
class Solution {
      public int solution(int n) {
          int answer = 0;
                    
          for(int i=2; i<=n; i++) {
              boolean flag = true;
              for(int j=2; j<=(int)Math.sqrt(i); j++) { //두 번째 방법에서는 j<i 부분을 j<Math.sqrt(i) 로 바꾼다.
                  if(i%j==0) {
                      flag = false;
                      break;
                  }
              }
              
              if(flag==true) answer++;
          }
          
          return answer;
      }
    }